### PR TITLE
Reducers and Actions

### PR DESCRIPTION
**Pull request by joyapisi**
## What I Did:
- Create a directory that contains all myRedux logic (/src/redux)
- Configure a Redux store (/src/redux/store.js)
- Define a slice of state for books that:
  - Includes an array of books (initial state: empty array)
  - Includes a reducer that adds a book
  - Includes a reducer that removes a book
- Define a slice of state for categories that:
  - Includes an array of categories (initial state: empty array)
  - Includes a reducer that checks the status and always returns "Under construction" (the initial state should check to that string)

### CODE REVIEW

File: src/redux/store.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { configureStore } from '@reduxjs/toolkit';
+import booksReducer from './books/booksSlice';
+
+const store = configureStore({
+  reducer: {
+    books: booksReducer,
+  },
+});
```
Comment (brainconnect93) [inline src/redux/store.js:9]: - [x] I would kindly suggest that you should import the `categoriesReducer` from categoriesSlice. Inorder to have a Reducer for it in the configureStore.

Possible implementation :point_down: 

```JS
import { configureStore } from "@reduxjs/toolkit"
import booksReducer from "./books/booksSlice"
import categoriesReducer from "./categories/categoriesSlice"

const store = configureStore({
  reducer: { books: booksReducer, categories: categoriesReducer },
})

export default store
```

**Review by brainconnect93 — CHANGES_REQUESTED** [review]
### Hi @joyapisi, 👋 

# STATUS: Changes Required :recycle:

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



### To Highlight :1st_place_medal: 

- Descriptive Pull Request. :heavy_check_mark:
- Action and Reducer functionality. :heavy_check_mark:
- Ducks pattern file structure. :heavy_check_mark:
- `@reduxjs/toolkit` & `react-redux` present in package.json :heavy_check_mark:
- Linters environment properly setup :heavy_check_mark:
- Good use of Git flow :heavy_check_mark:

### Changes Required

- [x]  _Please check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @brainconnect93 ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/store.js (hunk 1 above)
Comment (joyapisi) [reply src/redux/store.js:9]: Noted. I will correct that. Thank you.

**Review by mukhammadrizooff — APPROVED** [review]
Hi @joyapisi, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
- No linter errors ✔️ 
- Correct GitFlow ✔️ 
- Good implementations ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**